certora/ArrangerConduit.spec

```diff
@@ -0,0 +1,710 @@
+
+    assert what == to_bytes32(0x726f6c6573000000000000000000000000000000000000000000000000000000)
+           => rolesAfter == data, "file did not set roles";
+    assert what != to_bytes32(0x726f6c6573000000000000000000000000000000000000000000000000000000)
+           => rolesAfter == rolesBefore, "file did not keep unchanged roles";
+}
+
+// Verify revert rules on file
+rule file_address_revert(bytes32 what, address data) {
+    env e;
+
+    mathint wardsSender = wards(e.msg.sender);
+
+    file@withrevert(e, what, data);
+
+    bool revert1 = e.msg.value > 0;
+    bool revert2 = wardsSender != 1;
+    bool revert3 = what != to_bytes32(0x617272616e676572000000000000000000000000000000000000000000000000)
+                && what != to_bytes32(0x7265676973747279000000000000000000000000000000000000000000000000)
+                && what != to_bytes32(0x726f6c6573000000000000000000000000000000000000000000000000000000);
+
+    assert lastReverted <=> revert1 || revert2 || revert3, "Revert rules failed";
+}
+
+// Verify correct storage changes for non reverting setBroker
+rule setBroker(address usr, address asset, bool valid) {
+    env e;
+
+    address otherUsr;
+    require otherUsr != usr;
+
+    address otherAsset;
+    require otherAsset != asset;
+    
+    bool isBrokerOtherUsrBefore = isBroker(otherUsr, asset);
+    bool isBrokerOtherAssetBefore = isBroker(usr, otherAsset);
+    
+    setBroker(e, usr, asset, valid);
+
+    bool isBrokerUsrAfter = isBroker(usr, asset);
+    bool isBrokerOtherUsrAfter = isBroker(otherUsr, asset);
+    bool isBrokerOtherAssetAfter = isBroker(usr, otherAsset);
+
+    assert isBrokerUsrAfter == valid, "setBroker did not set broker to valid";
+    assert isBrokerOtherUsrAfter == isBrokerOtherUsrBefore, "setBroker did not keep unchanged the rest of brokers for the same asset";
+    assert isBrokerOtherAssetAfter == isBrokerOtherAssetBefore, "setBroker did not keep unchanged the rest of brokers for other assets";
+}
+
+// Verify revert rules on setBroker
+rule setBroker_revert(address usr, address asset, bool valid) {
+    env e;
+
+    mathint wardsSender = wards(e.msg.sender);
+
+    setBroker@withrevert(e, usr, asset, valid);
+
+    bool revert1 = e.msg.value > 0;
+    bool revert2 = wardsSender != 1;
+
+    assert revert1 => lastReverted, "revert1 failed";
+    assert revert2 => lastReverted, "revert2 failed";
+    assert lastReverted => revert1 || revert2, "Revert rules are not covering all the cases";
+}
+
+// Verify correct storage changes for non reverting deposit
+rule deposit(bytes32 ilk, address asset, uint256 amount) {
+    env e;
+
+    require asset == gem;
+
+    address buffer = registry.buffers(ilk);
+    require buffer != currentContract;
+
+    mathint balanceOfBufferBefore = gem.balanceOf(buffer);
+    mathint balanceOfConduitBefore = gem.balanceOf(currentContract);
+    mathint depositsBefore = deposits(asset, ilk);
+    mathint totalDepositsBefore = totalDeposits(asset);
```

Comment:
Following the pattern that we try to check every storage slot, we should be checking deposits for other assets or ilks and totalDeposits for other assets. Important to notice here is that we should be decoupling the "other" variable used for asset in deposits than the one using for total (if you follow the way I handle these cases mentioned above).
So I picture something like this:
```
address otherIlk;
address otherAsset;
require otherIlk != ilk || otherAsset != asset;

address otherAsset2;
require otherAsset2 != asset;

mathint depositsOtherBefore = deposits(otherAsset, otherIlk);
mathint totalDepositsOtherBefore = totalDeposits(otherAsset2);
```
This way we can ensure that all possibilities for both mappings can be checked.

Reply:
Ok, done more or less that, also in other places.